src/components/rangeslider/draw.js

```diff
@@ -286,7 +286,10 @@ function setDataRange(rangeSlider, gd, axisOpts, opts) {
         dataMax = clamp(opts.p2d(opts._pixelMax));
 
     window.requestAnimationFrame(function() {
-        Registry.call('relayout', gd, axisOpts._name + '.range', [dataMin, dataMax]);
+        var update = {};
+        update[axisOpts._name + '.range[0]'] = dataMin;
+        update[axisOpts._name + '.range[1]'] = dataMax;
+        Registry.call('_guiRelayout', gd, update);
```

Comment:
In an earlier iteration I was requiring that all update values be primitives - no arrays or objects, because there could be conflicts between this and other updates that do break it down. For example, first you update `xaxis.range` and store its initial value as the `preGUI` value; then later you update `xaxis.range[0]`, we see that this doesn't have a `preGUI` value so we store it - but what we're actually storing is the previous *post*GUI value (and anyway we have two conflicting values stored). However now this is taken care of in [`storeCurrent`](https://github.com/plotly/plotly.js/pull/3236/commits/a11ec44f388741dd21046d08a782e4e76fc8ba22#diff-2941ab69a12080c0633ff4ac8ea3aa83R1419) so we could in principle put this call back to the more compact form it had before.

Reply:
Thinking about this again, I think I would prefer reverting to the old `'xaxis.range': [/* */]` update to preserve backward compatibility. We should add an item about in [`v2`](#420) checklist though.

Reply:
Thanks! Resolved.